File: 백준/Silver/3986. 좋은 단어/좋은 단어.py
N = int(input())
cnt = 0
for _ in range(N):
    stack = []
    lst = list(input())
    for x in lst:
        if not stack:
            stack.append(x)
        elif stack[-1] == x:
            stack.pop()
        else:
            stack.append(x)
    if not stack:
        cnt += 1
print(cnt)

File: 백준/Silver/3986. 좋은 단어/좋은 단어.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayDeque;
import java.util.Deque;

public class Main {
	public static void main(String[] args) throws IOException {
			BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
			int N = Integer.parseInt(br.readLine());
			int cnt = 0;

			for (int i = 0; i < N; i++) {
				Deque<Character> stack = new ArrayDeque<>();
				String input =  br.readLine();

				for (char x : input.toCharArray()) {
					if (!stack.isEmpty() && stack.getLast() == x) {
						stack.pollLast();
						continue;
					}
					stack.addLast(x);
				}
				if (stack.isEmpty()) cnt++;
			}

			System.out.println(cnt);
	}
}
